vowpalwabbit/parser.cc

```diff
@@ -586,6 +588,14 @@ IGNORE_DEPRECATED_USAGE_END
 
         all.p->decision_service_json = input_options.dsjson;
       }
+      else if (input_options.flatbuffer)
+      {
+        all.flat_converter = new VW::parsers::flatbuffer::parser(all.data_filename);
+        all.flat_converter->init();
+
+        all.p->reader = VW::parsers::flatbuffer::flatbuffer_to_examples;
+        all.p->text_reader = VW::parsers::flatbuffer::read_flatbuffer;
```

Comment:
You could omit setting `all.p->text_reader` here which is set by default above to `VW::read_lines` since you don't need it. You should then be able to remove the `read_flatbuffer` function from `parse_flat_example` which doesn't do anything